sklearn/inspection/_plot/decision_boundary.py

```diff
@@ -0,0 +1,205 @@
+        estimator.__class__.__name__))
+
+
+class DecisionBoundaryDisplay:
+    """Decisions Boundary visualization.
+
+    It is recommend to use :func:`~sklearn.inspection.plot_decision_boundary`
+    to create a :class:`DecisionBoundaryDisplay`. All parameters are stored as
+    attributes.
+
+    Read more in the :ref:`User Guide <visualizations>`.
+
+    Parameters
+    ----------
+    xx0 : ndarray of shape (grid_resolution, grid_resolution)
+        First output of :func:`meshgrid <numpy.meshgrid>`.
+
+    xx1 : ndarray of shape (grid_resolution, grid_resolution)
+        Second output of :func:`meshgrid <numpy.meshgrid>`.
+
+    response : ndarray of shape (grid_resolution, grid_resolution)
+        Values of the response function.
+
+    Attributes
+    ----------
+    surface_ : matplotlib `QuadContourSet` or `QuadMesh`
+        If `plot_method` is 'contour' or 'contourf', `surface_` is a
+        :class:`QuadContourSet <matplotlib.contour.QuadContourSet>`. If
+        `plot_method is `pcolormesh`, `surface_` is a
+        :class:`QuadMesh <matplotlib.collections.QuadMesh>`.
+
+    ax_ : matplotlib Axes
+        Axes with confusion matrix.
+
+    figure_ : matplotlib Figure
+        Figure containing the confusion matrix.
+    """
+    def __init__(self, xx0, xx1, response):
+        self.xx0 = xx0
+        self.xx1 = xx1
+        self.response = response
+
+    def plot(self, plot_method='contourf', ax=None, **kwargs):
+        """Plot visualization.
+
+        Parameters
+        ----------
+        plot_method : {'contourf', 'contour', 'pcolormesh'}, default='contourf'
+            Plotting method to call when plotting the response. Please refer
+            to the following matplotlib documentation for details:
+            :func:`contourf <matplotlib.pyplot.contourf>`,
+            :func:`contour <matplotlib.pyplot.contour>`,
+            :func:`pcolomesh <matplotlib.pyplot.pcolomesh>`.
+
+        ax : Matplotlib axes, default=None
+            Axes object to plot on. If `None`, a new figure and axes is
+            created.
+
+        **kwargs : dict
+            Additional keyword arguments to be pased to the `plot_method`.
+
+        Returns
+        -------
+        display: :class:`~sklearn.inspection.DecisionBoundaryDisplay`
+        """
+        check_matplotlib_support('DecisionBoundaryDisplay.plot')
+        import matplotlib.pyplot as plt  # noqa
+
+        if plot_method not in ('contourf', 'contour', 'pcolormesh'):
+            raise ValueError("plot_method must be 'contourf', 'contour', or "
+                             "'pcolormesh'")
+
+        if ax is None:
+            _, ax = plt.subplots()
+
+        plot_func = getattr(ax, plot_method)
+        self.surface_ = plot_func(self.xx0, self.xx1, self.response, **kwargs)
+        self.ax_ = ax
+        self.figure_ = ax.figure
+        return self
+
+
+def plot_decision_boundary(estimator, X, grid_resolution=100, eps=1.0,
```

Comment:
I am a bit uncomfortable to add this function in the inspection module because I don't think that anyone should use this for inspecting or draw any conclusion on a subset of features. However, I see the benefit of having this function as a miscellaneous function for making example less complex and for teaching.

Reply:
To clarify, are you suggesting the `sklearn.utils` module? I am +0.5 on that.

The counter argument would be "Since we are using it to teach, there must be some conclusions we can draw from it...otherwise we wont use it to teach"

I kind of loosely defined inspection as "things you can do after your model has been fitted".

Reply:
> To clarify, are you suggesting the sklearn.utils module? I am +0.5 on that.

I don't know what would be the best module to accommodate it. `utils` seems the most suitable with the current tree-view for me. Otherwise, it would be in a new module.

>The counter argument would be "Since we are using it to teach, there must be some conclusions we can draw from it...otherwise we wont use it to teach" I kind of loosely defined inspection as "things you can do after your model has been fitted".

I personally used it to give intuitions about how an algorithm works in low dimensional space. For instance, it makes it easy to understand that our tree finds the best split along a specific feature (perpendicular split to an axis) or what is the meaning of `intercept_` or `coef_` in linear models.

I do not consider those as inspection/interpretation (because you should not use it in practice :)). Indeed, I am scared of the following: by making it easy, one can select a pair of features and make some kind of interpretation regarding what a model learnt.

On the opposite, I would not mind having the `plot_tree` in inspection because it depicts exactly what your model learnt.